crates/partition-store/src/partition_store_manager.rs

```diff
@@ -223,14 +265,31 @@ impl PartitionStoreManager {
     }
 
     pub async fn drop_partition(&self, partition_id: PartitionId) {
-        let mut guard = self.lookup.lock().await;
+        let mut guard = self.partition_stores.lock().await;
         self.rocksdb
             .inner()
             .as_raw_db()
             .drop_cf(&cf_for_partition(partition_id))
             .unwrap();
 
-        guard.live.remove(&partition_id);
+        guard.remove(&partition_id);
+        self.persisted_lsns.remove(&partition_id);
+    }
+
+    #[cfg(test)]
```

Comment:
Perhaps we can call this after shutting down a partition processor as well.

Reply:
Created a new https://github.com/restatedev/restate/issues/3251 for this.